stacker/blueprints/base.py

```diff
@@ -402,6 +405,36 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def parse_user_data(self, raw_userdata):
+        """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+        """
+        pattern = re.compile(r'{{([::|\w]+)}}')
+        userdata = ""
+        s_index = 0
+        variables = self.get_variables()
+
+        for match in pattern.finditer(raw_userdata):
+            userdata += raw_userdata[s_index:match.start()]
+
+            key = match.group(1)
+
+            if key in variables:
+                userdata += variables[key]
```

Comment:
one thing here is this would break if something was a `CFNString` since the value has to be `Ref`'d i can't remember exactly what that looks like off the top of my head, but i would add a test case to be sure.

one thing that would be cool is if this handled adding the `Ref` for you similar to the lookup.